qiskit_experiments/data_processing/data_processor.py

```diff
@@ -69,48 +103,79 @@ def is_trained(self) -> bool:
 
         return True
 
-    def __call__(self, data: Union[Dict, List[Dict]], **options) -> Tuple[Any, Any]:
+    def __call__(self, data: Union[Dict, List[Dict]], **options) -> Tuple[np.ndarray, np.ndarray]:
         """
         Call self on the given datum. This method sequentially calls the stored data actions
         on the datum.
 
         Args:
-            data: The data, typically from ExperimentData.data(...), that needs to be processed.
-            This dict or list of dicts also contains the metadata of each experiment.
+            data: The data, typically from ``ExperimentData.data(...)``,
+                that needs to be processed. This dict or list of dicts also contains
+                the metadata of each experiment.
             options: Run-time options given as keyword arguments that will be passed to the nodes.
 
         Returns:
             processed data: The data processed by the data processor.
         """
-        return self._call_internal(data, **options)
+        # Convert into conventional data format, This is temporary code block.
+        # Once following steps support ufloat array, data will be returned as-is.
+        # TODO need upgrade of following steps, i.e. curve fitter
```

Comment:
What is meant by this? Is this something to fix in this PR?

Reply:
As mentioned above. I wanted to split the PR into two pieces namely
- update of data processor
- update of curve analysis
In the first step (this PR), internal logic of dp is updated while curve analysis remains unchanged. Since curve analysis expects `Tuple[ndarray, ndarray]` we cannot return data as uarray. In the next PR, I'll update curve analysis to support ufloat, then this logic will be completely removed and the dp just returns data as-is.

Reply:
How big is the change to curve analysis? If its not too big it might be worthwhile having it in this PR to avoid these TODOs.

Reply:
Good point. Done in https://github.com/Qiskit/qiskit-experiments/pull/481/commits/006964f7890da91aab513af28b904f6e767a892f